chipsec/chipset.py

```diff
@@ -371,17 +371,40 @@ def use_native_api(self):
     ##################################################################################
 
     def init_xml_configuration( self ):
+        # Create a sorted config file list (xml only)
+        _cfg_files = []
         _cfg_path = os.path.join( chipsec.file.get_main_dir(), 'chipsec/cfg' )
-        # Load chipsec/cfg/common.xml configuration XML file common for all platforms if it exists
-        self.init_cfg_xml( os.path.join(_cfg_path,'common.xml'), self.code )
-        # Load chipsec/cfg/<code>.xml configuration XML file if it exists for platform <code>
+        for root, subdirs, files in os.walk(_cfg_path):
+            _cfg_files.extend([os.path.join(root, x) for x in files if fnmatch.fnmatch(x, '*.xml')])
+        _cfg_files.sort()
+        if logger().VERBOSE:
+            logger().log("[*] Configuration Files:")
+            for _xml in _cfg_files:
+                logger().log("[*] - {}".format(_xml))
+
+        # Load chipsec/cfg/common*.xml configuration XML file common for all platforms if it exists
+        loaded_files = []
+        if logger().VERBOSE: logger().log("[*] Loading Common Configurations:")
+        for _xml in _cfg_files:
+            if fnmatch.fnmatch(os.path.basename(_xml), 'common*.xml'):
+                loaded_files.append(_xml)
+                self.init_cfg_xml(_xml, self.code)
+
+        # Load chipsec/cfg/<code>*.xml configuration XML file if it exists for platform <code>
+        if logger().VERBOSE: logger().log("[*] Loading Platform Configurations:")
         if self.code and '' != self.code:
-            self.init_cfg_xml( os.path.join(_cfg_path,('%s.xml'%self.code)), self.code )
+            for _xml in _cfg_files:
+                if fnmatch.fnmatch(os.path.basename(_xml), '{}*.xml'.format(self.code)):
+                    loaded_files.append(_xml)
+                    self.init_cfg_xml(_xml, self.code)
+
         # Load configuration from all other XML files recursively (if any)
-        for dirname, subdirs, xml_fnames in os.walk( _cfg_path ):
-            for _xml in xml_fnames:
-                if fnmatch.fnmatch( _xml, '*.xml' ) and not fnmatch.fnmatch( _xml, 'common.xml' ) and not (_xml in ['%s.xml' % c.lower() for c in Chipset_Code]):
-                    self.init_cfg_xml( os.path.join(dirname,_xml), self.code )
+        if logger().VERBOSE: logger().log("[*] Loading Other Configurations:")
+        for plat in [c.lower() for c in Chipset_Code]:
+            loaded_files.extend([x for x in _cfg_files if fnmatch.fnmatch(os.path.basename(x), '{}*.xml'.format(plat))])
```

Comment:
It seems to me that it's better to not extend loaded_files here with <plat>*.xml files which weren't really loaded. Maybe use a separate list exclude_files in addition to loaded_files? The reason is that later we may want to list all loaded XML files and we would use loaded_files list for that and it would contain XML files which weren't really loaded.